cot-cli/src/migration_generator.rs

```diff
@@ -124,22 +123,22 @@ impl MigrationGenerator {
         }
     }
 
-    pub fn generate_migrations_as_source(&self) -> anyhow::Result<MigrationAsSource> {
+    pub fn generate_migrations_as_source(&self) -> anyhow::Result<Option<MigrationAsSource>> {
         let source_files = self.get_source_files()?;
         self.generate_migrations_as_source_from_files(source_files)
     }
 
     pub fn generate_migrations_as_source_from_files(
         &self,
         source_files: Vec<SourceFile>,
-    ) -> anyhow::Result<MigrationAsSource> {
-        if let Some(migration) = self.generate_migrations_as_generated_from_files(source_files)? {
-            let migration_name = migration.migration_name.clone();
-            let content = self.generate_migration_file_content(migration);
-            Ok(MigrationAsSource::new(migration_name, content))
-        } else {
-            bail!("unable to generate migrations from source files")
-        }
+    ) -> anyhow::Result<Option<MigrationAsSource>> {
+        Ok(self
+            .generate_migrations_as_generated_from_files(source_files)?
+            .map(|migration| {
+                let migration_name = migration.migration_name.clone();
+                let content = self.generate_migration_file_content(migration);
+                MigrationAsSource::new(migration_name, content)
+            }))
```

Comment:
IMHO, `Ok` wrapping such big blocks is hard to read and confusing
```suggestion
        let migrations = self
            .generate_migrations_as_generated_from_files(source_files)?
            .map(|migration| {
                let migration_name = migration.migration_name.clone();
                let content = self.generate_migration_file_content(migration);
                MigrationAsSource::new(migration_name, content)
            });
         Ok(migrations)
```